adr/adr.001.folder structure/adr.001.folder structure.md

```diff
@@ -0,0 +1,260 @@
+
+- `types`: TypeScript types and interfaces relevant to the feature.
+
+- `actors`: State machines or actors that control feature behavior.
+
+
+## Considered Options
+
+1. **Keep the current structure (organized by file type)**
+2. **Reorganize code by features (package by feature)**
+3. **Gradual approach (combination of file type and feature-based organization, working towards a package by feature structure)**
+
+## Pros and Cons of the Options
+
+### Option 1: Keep the current structure
+
+*Good, because:*
+
+- Familiar to current developers
+- No immediate refactoring needed
+
+*Bad, because:*
+
+- Scattered codebase makes it hard to find related files
+- Reduces productivity with AI-assisted tools
+- Difficult to maintain as the codebase grows
+
+### Option 2: Reorganize code by features
+
+*Good, because:*
+
+- Groups related code together, improving maintainability
+- Enhances productivity with AI-assisted tooling
+- Easier onboarding for new developers
+
+*Bad, because:*
+
+- Significant upfront refactoring effort
+- Potential for merge conflicts during transition
+- Requires updating import paths throughout the codebase
+
+### Option 3: Gradual approach
+
+*Good, because:*
+
+- Allows gradual transition
+- Balances familiarity with improved organization
+- Minimizes disruption to ongoing development
+
+*Bad, because:*
+
+- Can lead to inconsistent structure during transition
+- Might still scatter related code temporarily
+- Requires ongoing attention to migration
+
+## Proposed Option: Option 3 - Gradual approach
+
+We chose to reorganize the codebase by features gradually
+because it offers the benefits of improved organization
+while being more feasible to implement. A complete
+refactoring at once is impractical due to the size of our
+codebase and ongoing development needs. This approach allows
+us to incrementally improve our codebase structure as we
+touch different parts of the code.
+
+## Implementation Details
+
+- **Step 1:** Create a `features/` directory at the root of
+  the project.
+- **Step 2:** Create a `shared/` directory for common
+  components, hooks, utils, and other shared resources.
+- **Step 3:** As new features are developed, implement them
+  within the new structure under `features/`.
+- **Step 4:** When existing code is modified:
+  a. Assess if it belongs to a specific feature.
+  b. If so, move it to the appropriate feature directory.
+  c. If not, consider moving it to the `shared/` directory.
+- **Step 5:** Update import paths for moved files and their
+  dependencies.
+- **Step 6:** Gradually refactor existing features during
+  planned refactoring sprints or as part of regular
+  maintenance.
+- **Step 7:** Regularly review and update documentation to
+  reflect the evolving folder structure.
+- **Step 8:** Conduct team training sessions to ensure all
+  developers understand and follow the new structure.
+
+## File Naming Conventions
+
+We will adopt a consistent naming convention for key files
```

Comment:
This is quite opinionated but has been very successful in the past for teams I've worked on. It makes it as easy as possible to jump to files quickly from all IDEs (which is one of the most common things we do as devs)